### DISCUSSION TITLE
Side-by-side Decompiler View For Version Tracking

### DISCUSSION THREAD

**Discussion by clearbluejar** [NationalSecurityAgency/ghidra#5986] [Ideas]
@ghidracadabra 

Just saw this and wanted to ask about this commit. 
https://github.com/NationalSecurityAgency/ghidra/commit/04a932bab28389ca4ed04e5e550150b305f69f29#diff-ad066bfd4615b614255f4234de01384980ff2fbd76a0a8108506f85987a8c06d

Ghidra has for a long time had a side by side diff view (or function comparison provider) for the listing view:

![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/c1d79a4e-4c3e-4c6f-be97-557310a9ed45)

And has long been missing the side-by-size diff view  for decompilation. 

![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/8bc0af3e-db44-4b23-b438-1ebd13f27439)

I tried last nights [build](https://github.com/roblabla/ghidra-ci/releases/tag/2023-11-21) to see if this feature was available, but I didn't have much luck making it work. The screen shots above were from build 
![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/a27dd80d-1627-447d-96e3-de0bd64c2454)

Are these changes to support the decompiled diff view in the "function comparison provider" **and in VT**?

The listing side be side has always worked for VT, but is missing is this decompiled diff view.

![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/15ed838a-f654-4b2e-b198-155d8625b232)

In VT the decompiled side by side option is there, but the diff is not enabled. Do these new code changes enable this eventually?

Also! While on the topic! This would be really handy.

Add right click side by side diff options here:
![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/2d12dcf0-c586-43c4-b543-2b81c49d677a)

and here:

![image](https://github.com/NationalSecurityAgency/ghidra/assets/3752074/1bea9547-d471-40e7-a31c-2ba0c6453078)

Happy to submit each one of these requests official issues, but wanted to start the discussion first.

Potential issues/feature request summary:

1. Enable side by side decompile diff view for function comparison provider. (maybe this is what your commit already did, but not in last night build??)
2. Enable side by side decompile diff view for version tracking (like it is enabled for listing diffs)
3. Create a context menu option in the matches table that will display a side by side view in a new pop-up
4. Create a right click context menu option in the VT matches table in the code browser context that will display a side by side view in a new pop-up

**Comment by ghidracadabra** [comment]
@clearbluejar This commit adds a Ctrl-F "find" action to the dual decompiler view, and the ability to toggle between a horizontal split and a vertical split.

That said, we are actively working in this area.  It's probably worth submitting each of these requests as a separate issue.
